examples/pcr/quote_paramenc.c

```diff
@@ -0,0 +1,298 @@
+    cmdIn.authSes.sessionType = TPM_SE_POLICY;
+    cmdIn.authSes.tpmKey = TPM_RH_NULL;
+    cmdIn.authSes.bind = TPM_RH_NULL;
+    cmdIn.authSes.symmetric.algorithm = TPM_ALG_XOR;
+    cmdIn.authSes.symmetric.keyBits.xorr = TPM_ALG_SHA256;
+    cmdIn.authSes.symmetric.mode.sym = TPM_ALG_NULL;
+    cmdIn.authSes.encryptedSalt.size = 0;
+    cmdIn.authSes.authHash = TPM_ALG_SHA256;
+    cmdIn.authSes.nonceCaller.size = TPM_SHA256_DIGEST_SIZE;
+    rc = TPM2_GetNonce(cmdIn.authSes.nonceCaller.buffer,
+                       cmdIn.authSes.nonceCaller.size);
+    if (rc < 0) {
+        printf("TPM2_GetNonce failed 0x%x: %s\n", rc,
+            TPM2_GetRCString(rc));
+        goto exit;
+    }
+    rc = TPM2_StartAuthSession(&cmdIn.authSes, &cmdOut.authSes);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("\nTPM2_StartAuthSession failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+        goto exit;
+    }
+    sessionHandle = cmdOut.authSes.sessionHandle;
+    printf("TPM2_StartAuthSession: sessionHandle 0x%x\n", (word32)sessionHandle);
+
+
+    /* set auth for using the AIK */
+    session[0].auth.size = sizeof(gUsageAuth)-1;
+    XMEMCPY(session[0].auth.buffer, gUsageAuth, session[0].auth.size);
+
+    /* set session for XOR parameter encryption of the TPM Command */
+    session[1].sessionHandle = sessionHandle;
+    session[1].sessionAttributes = TPMA_SESSION_decrypt | TPMA_SESSION_continueSession;
+    session[1].symmetric.algorithm = TPM_ALG_XOR;
+    session[1].symmetric.keyBits.xorr = TPM_ALG_SHA256;
+    session[1].authHash = TPM_ALG_SHA256;
+    session[1].auth.size = sizeof(gXorAuth)-1;
+    XMEMCPY(session[1].auth.buffer, gXorAuth, session[1].auth.size);
+    session[1].nonce.size = TPM_SHA256_DIGEST_SIZE;
+    rc = TPM2_GetNonce(session[1].nonce.buffer,
+                       session[1].nonce.size);
+    if (rc < 0) {
+        printf("TPM2_GetNonce failed\n");
+        goto exit;
+    }
+
+    printf("\n\tCommand with ParamEnc\n");
+
+
+    /* Prepare Quote request */
+    XMEMSET(&cmdIn.quoteAsk, 0, sizeof(cmdIn.quoteAsk));
+    XMEMSET(&cmdOut.quoteResult, 0, sizeof(cmdOut.quoteResult));
+    cmdIn.quoteAsk.signHandle = rsaKey.handle.hndl;
+    cmdIn.quoteAsk.inScheme.scheme = TPM_ALG_RSASSA;
+    cmdIn.quoteAsk.inScheme.details.rsassa.hashAlg = TPM_ALG_SHA256;
+    /* This specifies the size of the parameter that will be encrypted */
+    cmdIn.quoteAsk.qualifyingData.size = sizeof(userData);
+    /* This is the data that will be encrypted */
+    XMEMCPY(cmdIn.quoteAsk.qualifyingData.buffer, userData,
+            cmdIn.quoteAsk.qualifyingData.size);
+   /* Choose PCR for signing */
+    TPM2_SetupPCRSel(&cmdIn.quoteAsk.PCRselect, TPM_ALG_SHA256, pcrIndex);
+
+
+    /* Get the PCR measurement signed by the TPM using the AIK key */
+    rc = TPM2_Quote(&cmdIn.quoteAsk, &cmdOut.quoteResult);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("TPM2_Quote failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
```

Comment:
The example appears to return `0xb: Unknown`. Is that really an error or just positive return code?

Reply:
Because access to CTX fails, it returns 0xb that is not a TPM defined error. I will push my original code for this section to emerge the segfault again.